### PR TITLE
Close Multiple Issues

### LINKED ISSUE #70: Modify CheckLiveDelta to return a segment delta when necessary
This simplifies the logic in Previous Segment slightly. (It also affects the segment delta columns in Splits and Subsplits, but those aren't implement in ls-core yet.)

https://github.com/LiveSplit/LiveSplit/commit/1e258ab5ae530fd409b54865889b854f11eddd76

### PR DESCRIPTION
**Pull request by hallcristobal**
Closes #70

### CODE REVIEW

**Comment by CryZe** [comment]
Why did you close this?

**Comment by hallcristobal** [comment]
Had more on local that needed to be commited

**Comment by CryZe** [comment]
So are you reopening it at some point?

**Comment by hallcristobal** [comment]
in like 2 minutes lol

File: src/run/parser/composite.rs [hunk 1]
```diff
@@ -23,6 +23,8 @@ pub struct ParsedRun {
 }
 
 fn parsed(run: Run, kind: TimerKind) -> ParsedRun {
+    let mut run = run;
+    run.fix_splits();
```
Comment (CryZe) [inline src/run/parser/composite.rs]: I wonder if this is the appropriate way to fix the splits when "opening them". Maybe that's something LiveSplit One should do instead, or maybe whenever the Timer sets a Run? (cc @wooferzfg)

Comment (hallcristobal) [reply src/run/parser/composite.rs]: I was thinking that or possibly when they're done being parsed, in the actual parse function, however that would have the same result as this in the end. But having it in livesplit-core would solidify that they would be fixed regardless of implementation.

Comment (wooferzfg) [reply src/run/parser/composite.rs]: Yeah, I think it makes more sense for LSO to do it. Also, run.fix_splits() needs to be called before saving splits as well; I'm not sure if we do that already or not.

Comment (wooferzfg) [reply src/run/parser/composite.rs]: If livesplit-core handled fixing splits when parsing and before saving that might work well too.

**Comment by CryZe** [comment]
I think I want to experiment with fixing in LSO and then make that decision. So I think it would be best if you'd split that out so I can merge the rest.

**Comment by hallcristobal** [comment]
Updated to remove Fixing

**Pull request merged**